src/teleop_twist_joy.cpp

```diff
@@ -65,41 +70,66 @@ struct TeleopTwistJoy::Impl
  * \param nh NodeHandle to use for setting up the publisher and subscriber.
  * \param nh_param NodeHandle to use for searching for configuration parameters.
  */
-TeleopTwistJoy::TeleopTwistJoy(ros::NodeHandle* nh, ros::NodeHandle* nh_param)
+TeleopTwistJoy::TeleopTwistJoy(rclcpp::node::Node::SharedPtr & node)
 {
-  pimpl_ = new Impl;
-
-  pimpl_->cmd_vel_pub = nh->advertise<geometry_msgs::Twist>("cmd_vel", 1, true);
-  pimpl_->joy_sub = nh->subscribe<sensor_msgs::Joy>("joy", 1, &TeleopTwistJoy::Impl::joyCallback, pimpl_);
-
-  nh_param->param<int>("enable_button", pimpl_->enable_button, 0);
-  nh_param->param<int>("enable_turbo_button", pimpl_->enable_turbo_button, -1);
-
-  if (nh_param->getParam("axis_linear", pimpl_->axis_linear_map))
-  {
-    nh_param->getParam("axis_linear", pimpl_->axis_linear_map);
-    nh_param->getParam("scale_linear", pimpl_->scale_linear_map);
-    nh_param->getParam("scale_linear_turbo", pimpl_->scale_linear_turbo_map);
-  }
-  else
+  pimpl_ = new Impl();
+
+  pimpl_->node = node;
+
+  rmw_qos_profile_t custom_qos_profile;
+  custom_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+  custom_qos_profile.depth = 50;
+  custom_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_RELIABLE;
+  custom_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_VOLATILE;
```

Comment:
So what I found was that a very large depth (of, say 50) worked around the problem I sometimes run into where rclcpp::spin_some() goes into an infinite loop.  Until we fix the bug that causes the infinite loop, I've just been putting a large depth on all of the QoS settings to make sure I don't hit the bug.